### DISCUSSION TITLE
Cannot Make libvips working with NetVips and magic support

### DISCUSSION THREAD

**Discussion by avaver** [libvips/libvips#3781] [Help wanted]
Hi, I have spent 3 days already trying to make NetVips to work with bmp files in linux docker container with no luck so far. 
Initially I have been using NetVips with pre-compiled linux-x64 binaries.
Then I realized I need libvips library with magick support, so I have compiled it with `-Dmagick=enabled -Dmagick-package=GraphicsMagick`
I used this script https://github.com/kleisauke/libvips-packaging/blob/main/build/lin.sh with my modifications:
```
mkdir ${DEPS}/gm
$CURL https://sourceforge.net/projects/graphicsmagick/files/graphicsmagick/${VERSION_GRAPHICSMAGICK}/GraphicsMagick-${VERSION_GRAPHICSMAGICK}.tar.xz/download | tar xJC ${DEPS}/gm --strip-components=1
cd ${DEPS}/gm
./configure --host=${CHOST} --prefix=${TARGET} --enable-static --disable-shared --disable-dependency-tracking 
make install-strip
```

also I had to add ` -fopenmp` compiler option

It compiled succesfully, and my test C# program was working fine with my library on Ubuntu.
But when I try to use my compiled library in Debian-based docker container, it fails with `VipsForeignLoad: buffer is not in a known format`
I tried to install different packages in container to make it work (like imagemagick, graphicsmagick, even libvips-dev) and I got it working at some point but I didn't note after what step it started working, and now I cannot get it to work again...

So basically my question is what do I need to do to enable BMP support and use it from NetVips. Thank you!

**Comment by jcupitt** [comment]
Hi @avaver,

I'd ask on the netvips issue tracker: https://github.com/kleisauke/net-vips/issues

**Reply by avaver** [reply]
thank you for your prompt reply, will do!
Just to make sure - if I want libvips libraries with graphicsmagick support - is it sufficient to install libvips and graphicsmagick packages with `apt-get install libvips graphicsmagick --no-install-recommends -y` or I must build libvips from sources?

**Reply by jcupitt** [reply]
You need to install the `dev` package for image/graphicsmagick, then rebuild libvips from source. libvips makes direct calls into the magick library to improve speed, so libvips has to be compiled with all of the magick header files present.

netvips should have an option to install an imagemagick binary for you -- you need the `-all` variant. I'd check the netvips install documentation again.
